### PR TITLE
NDArray 1: germinate TNDArray and PNDArray

### PR DESCRIPTION
**Pull request by jbloom22**

### CODE REVIEW

File: hail/src/main/scala/is/hail/expr/types/TNDArray.scala [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+import org.apache.spark.sql.Row
+
+import scala.reflect.{ClassTag, _}
+
+final case class TNDArray(elementType: Type, override val required: Boolean = false) extends ComplexType {
+  val representation: Type = TStruct(
+    "flags" -> TInt64Required,           // encodes data layout
+    "shape" -> TArray(TInt64Required),   // length is ndim
+    "offset" ->  TInt64Required,         // offset into data
+    "strides" -> TArray(TInt64Required), // stride in each dim, length is 1
```
Comment (cseed) [inline hail/src/main/scala/is/hail/expr/types/TNDArray.scala]: Length is 1?  What's that mean?  Isn't the length ndim?

File: hail/src/main/scala/is/hail/expr/types/TNDArray.scala [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+    "data" -> TArray(elementType)
+  )
```
Comment (cseed) [inline hail/src/main/scala/is/hail/expr/types/TNDArray.scala]: I don't like this trailing ) on its own line business, but whatever.

Comment (jbloom22) [reply hail/src/main/scala/is/hail/expr/types/TNDArray.scala]: yeah, i'd copied that pattern but agree, will change later

File: hail/src/main/scala/is/hail/expr/types/TNDArray.scala (hunk 1 above)
Comment (jbloom22) [reply hail/src/main/scala/is/hail/expr/types/TNDArray.scala]: yes, meant `ndim`, that was a typo, will fix in next PR

**Pull request merged**